ts/src/program/accounts-resolver.ts

```diff
@@ -57,23 +64,9 @@ export class AccountsResolver<IDL extends Idl, I extends AllInstructions<IDL>> {
       }
 
       // Common accounts are auto populated with magic names by convention.
-      switch (accountDescName) {
-        case "systemProgram":
-          if (this._accounts[accountDescName] === undefined) {
-            this._accounts[accountDescName] = SystemProgram.programId;
-          }
-        case "rent":
-          if (this._accounts[accountDescName] === undefined) {
-            this._accounts[accountDescName] = SYSVAR_RENT_PUBKEY;
-          }
-        case "tokenProgram":
-          if (this._accounts[accountDescName] === undefined) {
-            this._accounts[accountDescName] = TOKEN_PROGRAM_ID;
-          }
-        case "associatedTokenProgram":
-          if (this._accounts[accountDescName] === undefined) {
-            this._accounts[accountDescName] = ASSOCIATED_PROGRAM_ID;
-          }
+      if (Reflect.has(AccountsResolver.CONST_ACCOUNTS, accountDescName)) {
```

Comment:
Should this be optional? I can't think of a good reason why someone would do this, but they could potentially use `systemProgram` with a different public key, in which case this wouldn't do what is expected.

Reply:
if it's optional then the type definitions of `.accounts()` needs to change. other than testing for failures (and even that's a bit of a stretch) i couldn't come up with a use-case for providing the wrong public key for global programs and accounts.

Reply:
I can change it so it's optional though. there's a discussion about this (sort of) in the linked issue.